torch/utils/data/dataset.py

```diff
@@ -174,6 +176,21 @@ def class_function(cls, source_dp, *args, **kwargs):
         function = functools.partial(class_function, cls_to_register)
         IterableDataset.functions[function_name] = function
 
+    # TODO: Determinin if force all IterableDataset and IterDataPipe with annotation
+    #  def __init_subclass__(cls, *args, **kwargs):
+    #      if not hasattr(cls, 'type'):
+    #          raise TypeError('Class {} needs to be specified with type'.format(cls.__name__))
```

Comment:
In case of BC-breaking, IMO it should be commented out. There are lots of Dataset not specifying Type.